framework/src/modules/network/network.js

```diff
@@ -0,0 +1,101 @@
+const {
+	P2P,
+	EVENT_REQUEST_RECEIVED,
+	EVENT_MESSAGE_RECEIVED,
+} = require('@liskhq/lisk-p2p');
+const { createLoggerComponent } = require('../../components/logger');
+
+/**
+ * Network Module
+ *
+ * @namespace Framework.modules.network
+ * @type {module.Network}
+ */
+module.exports = class Network {
+	constructor(options) {
+		this.options = options;
+		this.channel = null;
+		this.logger = null;
+	}
+
+	async bootstrap(channel) {
+		this.channel = channel;
+
+		const loggerConfig = await this.channel.invoke(
+			'lisk:getComponentConfig',
+			'logger'
+		);
+
+		this.logger = createLoggerComponent(loggerConfig);
+
+		const initialNodeInfo = await this.channel.invoke('chain:getNodeInfo');
+
+		const p2pConfig = {
+			...this.options,
+			nodeInfo: {
+				...initialNodeInfo,
+				wsPort: this.options.nodeInfo.wsPort,
+			},
+		};
+
+		this.p2p = new P2P(p2pConfig);
+
+		this._handleUpdateNodeInfo = event => {
+			this.p2p.applyNodeInfo(event.data);
+		};
+
+		this.channel.subscribe(
+			'chain:system:updateNodeInfo',
+			this._handleUpdateNodeInfo
+		);
+
+		this.p2p.on(EVENT_REQUEST_RECEIVED, async request => {
+			try {
+				const result = await this.channel.invoke(
+					request.procedure,
+					request.data
+				);
+				request.end(result); // Send the response back to the peer.
+			} catch (error) {
+				request.error(error); // Send an error back to the peer.
+			}
+		});
+
+		this.p2p.on(EVENT_MESSAGE_RECEIVED, async packet => {
+			this.channel.publish(`network:${packet.event}`, packet.data);
+		});
+
+		try {
```

Comment:
no error event handling?

Reply:
this can be in other issue if needed =)

Reply:
Ah yep, I added event handling and logging.